From: Lucas Stach <l.stach@pengutronix.de>
Date: Wed, 1 Feb 2023 15:31:46 +0100
Subject: [PATCH] WIP: arm64: dts: imx8mp: add NPU node

WIP Note: rebased on top of mainlined node. Diff is only assigned clocks

Add the DT node describing the neural network accelerator (NPU). The NPU
is derived from the VeriSilicon GPU design and as such can use the same
binding with the driver being able to deduce the specific core features
from a set of identification registers.

Signed-off-by: Lucas Stach <l.stach@pengutronix.de>
---
 arch/arm64/boot/dts/freescale/imx8mp.dtsi | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp.dtsi b/arch/arm64/boot/dts/freescale/imx8mp.dtsi
index 64e5d535fe2e..cda9ef87ce53 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp.dtsi
@@ -2188,6 +2188,13 @@ npu: npu@38500000 {
 				 <&clk IMX8MP_CLK_ML_AXI>,
 				 <&clk IMX8MP_CLK_ML_AHB>;
 			clock-names = "core", "shader", "bus", "reg";
+			assigned-clocks = <&clk IMX8MP_CLK_ML_CORE>,
+					  <&clk IMX8MP_CLK_ML_AXI>,
+					  <&clk IMX8MP_CLK_ML_AHB>;
+			assigned-clock-parents = <&clk IMX8MP_SYS_PLL1_800M>,
+						 <&clk IMX8MP_SYS_PLL1_800M>,
+						 <&clk IMX8MP_SYS_PLL1_800M>;
+			assigned-clock-rates = <800000000>, <800000000>, <400000000>;
 			power-domains = <&pgc_mlmix>;
 		};
 
